### PR TITLE
batch create: add --metrics-build-id flag

### PR DESCRIPTION
**Pull request by benabineri**
# Description of change

queued behind https://github.com/resim-ai/api-client/pull/55/files

## Checklist

- [ ] I have self-reviewed this change.
- [ ] I have tested this change.
- [ ] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -13,4 +13,5 @@
 
 - [ ] I have self-reviewed this change.
 - [ ] I have tested this change.
-- [ ] This change is covered by tests that are already landed, or in this PR.
\ No newline at end of file
+- [ ] This change is covered by tests that are already landed, or in this PR.
+- [ ] I have updated the changelog, if appropriate.
```
Comment (iainjwhiteside) [inline .github/pull_request_template.md:17]: ta

File: api/client.gen.go [hunk 2]
```diff
@@ -40,11 +40,27 @@ const (
 
 // Defines values for JobStatus.
 const (
-	JobStatusCANCELLED JobStatus = "CANCELLED"
-	JobStatusFAILED    JobStatus = "FAILED"
-	JobStatusRUNNING   JobStatus = "RUNNING"
-	JobStatusSUBMITTED JobStatus = "SUBMITTED"
-	JobStatusSUCCEEDED JobStatus = "SUCCEEDED"
+	JobStatusCANCELLED         JobStatus = "CANCELLED"
```
Comment (iainjwhiteside) [inline api/client.gen.go:43]: nice. let's make a note to regenerate this from the landed rerun.yml when that PR on the rerun side lands

File: testing/end_to_end_test.go [hunk 3]
```diff
@@ -1028,13 +1034,13 @@ func (s *EndToEndTestSuite) TestBatchAndLogs() {
...
-	output = s.runCommand(s.createBatch(buildIDString, []string{}, []string{}, []string{}, GithubFalse), ExpectError)
+	output = s.runCommand(s.createBatch(buildIDString, []string{}, []string{}, []string{}, "", GithubFalse), ExpectError)
 	s.Contains(output.StdErr, FailedToCreateBatch)
 	// Try a batch without a build id:
-	output = s.runCommand(s.createBatch("", []string{experienceIDString1, experienceIDString2}, []string{}, []string{}, GithubFalse), ExpectError)
+	output = s.runCommand(s.createBatch("", []string{experienceIDString1, experienceIDString2}, []string{}, []string{}, "", GithubFalse), ExpectError)
 	s.Contains(output.StdErr, InvalidBuildID)
 	// Try a batch with both experience tag ids and experience tag names (even if fake):
-	output = s.runCommand(s.createBatch(buildIDString, []string{}, []string{"tag-id"}, []string{"tag-name"}, GithubFalse), ExpectError)
+	output = s.runCommand(s.createBatch(buildIDString, []string{}, []string{"tag-id"}, []string{"tag-name"}, "", GithubFalse), ExpectError)
```
Comment (iainjwhiteside) [inline testing/end_to_end_test.go:1043]: I'll add a test that creates a batch with the metrics build id.

**Review by iainjwhiteside — APPROVED** [review]

File: testing/end_to_end_test.go (hunk 3 above)
Comment (benabineri) [reply testing/end_to_end_test.go:1043]: thanks, yeah the tests here aren't super comprehensive because I didn't have a way to create metrics builds :)

**Pull request merged**